### ISSUE TITLE
Building with ghc-9.2.1

### ISSUE THREAD

**Issue by locallycompact** [input-output-hk/cardano-base#245]
Hi, I'm trying to build this repository with ghc-9.2.1 and running in to these errors that I don't know how to approach.

```
cardano-crypto-class> /home/lc/Source/github.com/input-output-hk/cardano-base/cardano-crypto-class/src/Cardano/Crypto/PackedBytes.hs:313:9: error:
cardano-crypto-class>     • Couldn't match expected type ‘Word32#’ with actual type ‘Word#’
cardano-crypto-class>     • In the first argument of ‘W32#’, namely
cardano-crypto-class>         ‘(narrow32Word# (byteSwap32# (indexWord8ArrayAsWord32# ba# i#)))’
cardano-crypto-class>       In the expression:
cardano-crypto-class>         W32#
cardano-crypto-class>           (narrow32Word# (byteSwap32# (indexWord8ArrayAsWord32# ba# i#)))
cardano-crypto-class>       In an equation for ‘indexWord32BE’:
cardano-crypto-class>           indexWord32BE (ByteArray ba#) (I# i#)
cardano-crypto-class>             = W32#
cardano-crypto-class>                 (narrow32Word# (byteSwap32# (indexWord8ArrayAsWord32# ba# i#)))
cardano-crypto-class>     |
cardano-crypto-class> 313 |   W32# (narrow32Word# (byteSwap32# (indexWord8ArrayAsWord32# ba# i#)))
cardano-crypto-class>     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cardano-crypto-class>
cardano-crypto-class> /home/lc/Source/github.com/input-output-hk/cardano-base/cardano-crypto-class/src/Cardano/Crypto/PackedBytes.hs:313:37: error:
cardano-crypto-class>     • Couldn't match expected type ‘Word#’ with actual type ‘Word32#’
cardano-crypto-class>     • In the first argument of ‘byteSwap32#’, namely
cardano-crypto-class>         ‘(indexWord8ArrayAsWord32# ba# i#)’
cardano-crypto-class>       In the first argument of ‘narrow32Word#’, namely
cardano-crypto-class>         ‘(byteSwap32# (indexWord8ArrayAsWord32# ba# i#))’
cardano-crypto-class>       In the first argument of ‘W32#’, namely
cardano-crypto-class>         ‘(narrow32Word# (byteSwap32# (indexWord8ArrayAsWord32# ba# i#)))’
cardano-crypto-class>     |
cardano-crypto-class> 313 |   W32# (narrow32Word# (byteSwap32# (indexWord8ArrayAsWord32# ba# i#)))
cardano-crypto-class>     |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
cardano-crypto-class>
cardano-crypto-class> /home/lc/Source/github.com/input-output-hk/cardano-base/cardano-crypto-class/src/Cardano/Crypto/PackedBytes.hs:328:48: error:
cardano-crypto-class>     • Couldn't match expected type ‘Word32#’ with actual type ‘Word#’
cardano-crypto-class>     • In the third argument of ‘writeWord8ArrayAsWord32#’, namely
cardano-crypto-class>         ‘wbe#’
cardano-crypto-class>       In the first argument of ‘primitive_’, namely
cardano-crypto-class>         ‘(writeWord8ArrayAsWord32# mba# i# wbe#)’
cardano-crypto-class>       In the expression:
cardano-crypto-class>         primitive_ (writeWord8ArrayAsWord32# mba# i# wbe#)
cardano-crypto-class>     |
cardano-crypto-class> 328 |   primitive_ (writeWord8ArrayAsWord32# mba# i# wbe#)
cardano-crypto-class>     |                                                ^^^^
cardano-crypto-class>
cardano-crypto-class> /home/lc/Source/github.com/input-output-hk/cardano-base/cardano-crypto-class/src/Cardano/Crypto/PackedBytes.hs:333:40: error:
cardano-crypto-class>     • Couldn't match expected type ‘Word#’ with actual type ‘Word32#’
cardano-crypto-class>     • In the first argument of ‘byteSwap32#’, namely ‘w#’
cardano-crypto-class>       In the first argument of ‘narrow32Word#’, namely ‘(byteSwap32# w#)’
cardano-crypto-class>       In the expression: narrow32Word# (byteSwap32# w#)
cardano-crypto-class>     |
cardano-crypto-class> 333 |     !wbe# = narrow32Word# (byteSwap32# w#)
```

The constructor `W32#` has changed in base-4.16, from`Word#` to `Word32#`, but there doesn't seem to be any migration information on this so I'm not sure what to do.

**Comment by lehins** [comment]
Yeah, this was sort of expected. I'll submit a patch for this tomorrow.

**Comment by locallycompact** [comment]
If I could have some direction I'd be happy to make this change.

**Comment by tbidne** [comment]
Background [here](https://github.com/fumieval/Haskell-breaking-changes#ongoing-word8).

**Comment by locallycompact** [comment]
@tbidne Thanks! Made some progress.

**Comment by lehins** [comment]
@locallycompact Sorry, got sidetracked with other issues and holidays. The fix for this problem is in #248 If you could give it a try it would be awesome.

**Comment by locallycompact** [comment]
@lehins Thanks! I've added a few more https://github.com/input-output-hk/cardano-base/pull/249
